packages/checkbox/src/useCheckbox.ts

```diff
@@ -1,23 +1,183 @@
     elForm,
     ELEMENT,
-    _elFormItemSize,
+    elFormItemSize,
     elFormItem,
   }
 }
+
+const useModel = (props: ICheckboxProps) => {
+  let selfModel = false
+  const { emit } = getCurrentInstance()
+  const { isGroup, checkboxGroup } = useCheckboxGroup()
+  const isLimitExceeded = ref(false)
+  const store = computed(() => checkboxGroup ? checkboxGroup.modelValue?.value : props.modelValue)
+  const model = computed({
+    get() {
+      return isGroup.value
+        ? store.value
+        : props.modelValue ?? selfModel
+    },
+
+    set(val: unknown) {
+      if (isGroup.value && Array.isArray(val)) {
+        isLimitExceeded.value = false
+
+        if (checkboxGroup.min !== undefined && val.length < checkboxGroup.min.value) {
+          isLimitExceeded.value = true
+        }
+        if (checkboxGroup.max !== undefined && val.length > checkboxGroup.max.value) {
+          isLimitExceeded.value = true
+        }
+
+        isLimitExceeded.value === false && checkboxGroup.changeEvent?.(val)
+      } else {
+        emit(UPDATE_MODEL_EVENT, val)
+        selfModel = val as boolean
+      }
+    },
+  })
+
+  return {
+    model,
+    isLimitExceeded,
+  }
+}
+
+const useCheckboxStatus = (props: ICheckboxProps, { model }: PartialReturnType<typeof useModel>) => {
+  const { isGroup, checkboxGroup, elFormItemSize, ELEMENT } = useCheckboxGroup()
+  const focus = ref(false)
+  const size = computed<string|undefined>(() => checkboxGroup.checkboxGroupSize?.value || elFormItemSize.value || (ELEMENT || {}).size)
+  const isChecked = computed(() => {
+    const value = model.value
+    if (toTypeString(value) === '[object Boolean]') {
+      return Boolean(value)
+    } else if (Array.isArray(value)) {
+      return value.includes(props.label)
+    } else if (value !== null && value !== undefined) {
+      return value === props.trueLabel
+    }
+  })
+  const checkboxSize = computed(() => {
+    const temCheckboxSize = props.size || elFormItemSize.value || (ELEMENT || {} as any).size
+    return isGroup.value
+      ? checkboxGroup.checkboxGroupSize?.value || temCheckboxSize
+      : temCheckboxSize
+  })
+
+  return {
+    isChecked,
+    focus,
+    size,
+    checkboxSize,
+  }
+}
+
+const useDisabled = (
+  props: ICheckboxProps,
+  { model, isChecked }: PartialReturnType<typeof useModel> & PartialReturnType<typeof useCheckboxStatus>,
+) => {
+  const { elForm, isGroup, checkboxGroup } = useCheckboxGroup()
+  const isLimitDisabled = computed(() => {
+    const max = checkboxGroup.max?.value
+    const min = checkboxGroup.min?.value
+    return !!(max || min) && (model.value.length >= max && !isChecked.value) ||
+      (model.value.length <= min && isChecked.value)
+  })
+  const isDisabled = computed(() => {
+    return isGroup.value
+      ? checkboxGroup.disabled?.value || props.disabled ||  (elForm as any || {} as any).disabled?.value || isLimitDisabled.value
+      : props.disabled || (elForm as any || {} as any).disabled?.value
```

Comment:
This is some sort of disaster for both the chained short cut operator and type casting.
Also, when `isGroup.value` goes to the falsy branch and props.disabled is `false` also elForm has not been given by injection, the final result would be `undefined` instead of a `boolean` type that I assume you need here.